collect/central_collector.go

```diff
@@ -239,6 +239,13 @@ func (c *CentralCollector) Stop() error {
 	ctx, cancel := context.WithTimeout(ctx, c.Config.GetCollectionConfig().GetShutdownDelay())
 	defer cancel()
 
+	// we have to make sure the health check says we're alive but not accepting data during shutdown
+	c.Health.Unregister(receiverHealth)
+	c.Health.Unregister(deciderHealth)
+	// reregister the sender health check to a much longer time so we can finish sending traces
+	c.Health.Register(senderHealth, 5*time.Second)
+	c.Health.Ready(senderHealth, true)
```

Comment:
I'm sorry I probably wasn't clear. I think line 247 still needs to be in the `sendCycle.Run` so that the sender keeps checking in with the health check system

Reply:
You're absolutely right -- I pulled out a "shutdownMonitor" function into a local variable -- I think it's clearer because passing it inline to Run made the logic confusing to me.